### ISSUE TITLE
Conformance tests for BackendTLSPolicy

### ISSUE THREAD

**Issue by candita** [kubernetes-sigs/gateway-api#3138]
**What would you like to be added**:
Conformance tests for BackendTLSPolicy. Comment below if you're interested in working on covering any of these areas.

Core Capabilities:
- [ ]  Valid BackendTLSPolicy with 1 targetRef/service using CACertificateRef and matching hostname
- [ ]  Valid BackendTLSPolicy with 1 targetRef/service using WellKnownCACertificates and matching hostname

- [ ] Invalid: targetRef in different namespace
- [ ] Invalid: both CACertificateRef and WellKnownCACertificates is specified
- [ ] Invalid: hostname doesn't match the hostname in the certificate served by the backend
- [ ] Invalid: SNI hostname invalid - not RFC1123
- [ ] Invalid BackendTLSPolicy performs no default forwarding
- [ ] Name not set
- [ ] Kind not set
- [ ] Namespace (of targetRef) not set

**Why this is needed**:
This is needed in order for BackendTLSPolicy to graduate from v1alpha3 to v1.

**Comment by candita** [comment]
cc @whitneygriffith @mlavacca

**Comment by whitneygriffith** [comment]
#1897

**Comment by mlavacca** [comment]
> Invalid: both CACertificateRef and WellKnownCACertificates is specified

I think this test cannot be implemented, as this rule is directly enforced by CEL:

https://github.com/kubernetes-sigs/gateway-api/blob/d49ae960279b7022c8f7e041221df306ef64094d/apis/v1alpha3/backendtlspolicy_types.go#L81

**Comment by mlavacca** [comment]
> Valid BackendTLSPolicy with 1 targetRef/service using WellKnownCACertificates and matching hostname

`WellKnownCACertificates` is an implementation-specific feature, therefore I think we should either:
- leave it out of conformance tests, or
- accept that the implementation set the `Accepted` condition to false, as stated in https://github.com/kubernetes-sigs/gateway-api/blob/d49ae960279b7022c8f7e041221df306ef64094d/apis/v1alpha3/backendtlspolicy_types.go#L115-L118

If we go the second way, though, in my opinion, this is beyond the bare minimum set of conformance tests needed for graduation.

**Comment by mlavacca** [comment]
> Invalid: targetRef in different namespace

The TargetRef is a `LocalPolicyTargetReference`, there is no namespace field in it. I think that a `TargetRef` in a different namespace is impossible, given the current API state.

https://github.com/kubernetes-sigs/gateway-api/blob/86c06a0b62587a627a9929d5943c5894a608ad62/apis/v1alpha2/policy_types.go#L39-L48

**Comment by whitneygriffith** [comment]
> > Invalid: targetRef in different namespace
> 
> The TargetRef is a `LocalPolicyTargetReference`, there is no namespace field in it. I think that a `TargetRef` in a different namespace is impossible, given the current API state.
> 
> https://github.com/kubernetes-sigs/gateway-api/blob/86c06a0b62587a627a9929d5943c5894a608ad62/apis/v1alpha2/policy_types.go#L39-L48

I agree, we don't need a conformance test for this case. The same applies for Invalid: Namespace (of targetRef) not set.

**Comment by whitneygriffith** [comment]
Updated test cases: 

- [ ] Valid BackendTLSPolicy with 1 targetRef/service using CACertificateRef and matching hostname
- [ ]  Valid BackendTLSPolicy with 1 targetRef/service using WellKnownCACertificates and matching hostname
- [x]  Invalid: targetRef in different namespace (targetRef is of type [LocalPolicyTargetReference](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1alpha2.LocalPolicyTargetReference) which does not have a namespace field)
- [x]  Invalid: both CACertificateRef and WellKnownCACertificates is specified (enforced by CEL)
- [ ]  Invalid: hostname doesn't match the hostname in the certificate served by the backend
- [x]  Invalid: SNI hostname invalid - not RFC1123 (should be caught by validation)
- [ ]  Invalid BackendTLSPolicy performs no default forwarding
- [x]  Invalid: Namespace (of targetRef) not set (targetRef is of type [LocalPolicyTargetReference](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1alpha2.LocalPolicyTargetReference) which does not have a namespace field)
